sympy/solvers/bivariate.py

```diff
@@ -196,6 +277,41 @@ def _solve_lambert(f, symbol, gens):
       a = -1, d = a*log(p), f = -log(d) - g*log(p)
     """
 
+    def _solve_even_degree_expr(expr, symbol):
+        """
+        Helper function to solve an expr containing dummy variable t at even
+        degree places to replace them with +symbol and -symbol.
+
+        For example:
+        Equation 2*log(t) + x/2 - log(a) = 0 will return solutions
+        of both of these equations.
+        1. 2*log(x) + x/2 - log(a) = 0
+        2. 2*log(-x) + x/2 - log(a) = 0
+
+        Parameters
+        ==========
+
+        expr : Expr
+            The expression which includes a dummy variable t which can be
+            replaced with +symbol and -symbol.
+
+        symbol : Symbol
+            The symbol concerned in expr.
+
+        Returns
+        =======
+
+        Solution of two equations generated by replacing t(at even degree places)
+        with positive and negative symbol.
+
+        """
+        plhs, nlhs = map(lambda sym: expr.xreplace({t: sym}), (symbol, -symbol))
```

Comment:
💭 that's interesting. I usually think of map as changing an expression. In this case the thing you are mapping onto are substitution values. It works. It's just a little like listening to Yoda talk when your native language is SVO instead of SOV order. Perhaps `plhs, nlhs = [expr.xreplace(t: s*sym) for s in (1, -1)]` would be more transparent?

Reply:
> It's just a little like listening to Yoda talk when your native language is SVO instead of SOV order. 

You always surprise by your knowledge :+1: . I didn't know this thing before.

>Perhaps plhs, nlhs = [expr.xreplace(t: s*sym) for s in (1, -1)] would be more transparent?

Yeah I feel so.